source/iterator/cdc_iterator_test.go

```diff
@@ -0,0 +1,168 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iterator
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/conduitio/conduit-connector-zendesk/source/position"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/tomb.v2"
+)
+
+func TestNewCDCIterator(t *testing.T) {
+	tests := []struct {
+		name          string
+		Domain        string
+		UserName      string
+		APIToken      string
+		PollingPeriod time.Duration
+		tp            position.TicketPosition
+		isError       bool
+	}{
+		{
+			name:          "NewCDCIterator with lastModifiedTime=0",
+			Domain:        "testlab",
+			UserName:      "test@testlab.com",
+			APIToken:      "gkdsaj)({jgo43646435#$!ga",
+			PollingPeriod: time.Millisecond,
+			tp:            position.TicketPosition{LastModified: time.Time{}},
+		}, {
+			name:          "NewCDCIterator with lastModifiedTime=2022-01-02T15:04:05Z",
+			Domain:        "testlab",
+			UserName:      "test@testlab.com",
+			APIToken:      "gkdsaj)({jgo43646435#$!ga",
+			PollingPeriod: time.Millisecond,
+			tp: position.TicketPosition{
+				LastModified: time.Date(2022, 01, 02,
+					15, 04, 05, 0, time.UTC),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			res, err := NewCDCIterator(context.Background(), tt.UserName, tt.APIToken, tt.Domain, tt.PollingPeriod, tt.tp)
+			if tt.isError {
+				assert.NotNil(t, err)
+			} else {
+				assert.NotNil(t, res)
+				assert.NotNil(t, res.caches)
+				assert.NotNil(t, res.buffer)
+				assert.NotNil(t, res.tomb)
+				assert.NotNil(t, res.ticker)
+				expectedTime := tt.tp.LastModified.Unix()
+				if expectedTime < 0 {
+					expectedTime = 0
+				}
+				assert.Equal(t, expectedTime, res.lastModifiedTime.Unix())
+			}
+		})
+	}
+}
+
+func TestFlush(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	tmbWithCtx, _ := tomb.WithContext(ctx)
+	cdc := &CDCIterator{
+		buffer: make(chan sdk.Record, 1),
+		caches: make(chan []sdk.Record, 1),
+		tomb:   tmbWithCtx,
+	}
+	randomErr := errors.New("random error")
+	cdc.tomb.Go(cdc.flush)
+
+	in := sdk.Record{Position: []byte("some_position")}
+	cdc.caches <- []sdk.Record{in}
+	for {
+		select {
+		case <-cdc.tomb.Dying():
+			assert.EqualError(t, cdc.tomb.Err(), randomErr.Error())
+			cancel()
+			return
+		case out := <-cdc.buffer:
+			assert.Equal(t, in, out)
+			cdc.tomb.Kill(randomErr)
+		}
+	}
+}
+
+func TestNext(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	tmbWithCtx, _ := tomb.WithContext(ctx)
+	cdc := &CDCIterator{
```

Comment:
IMHO, we should instantiate structs in tests in the same way we do that in production code. We should also interact with it only through its public API (i.e. exported fields and methods). Unexported fields are part of the internal impl. and we should rely on that, since it can change, but the behavior shouldn't change (which is what tests need to verify).

Reply:
@hariiwill in case you missed it ^

Reply:
Done